src/enumeration.rs

```diff
@@ -314,6 +314,9 @@ impl std::fmt::Debug for DeviceInfo {
         #[cfg(target_os = "linux")]
         {
             s.field("sysfs_path", &self.path);
+        }
+        #[cfg(any(target_os = "linux", target_os = "android"))]
+        {
```

Comment:
I missed this, but @tuna-f1sh [noticed](https://github.com/kevinmehall/nusb/pull/77#issuecomment-2368027477) when trying to rebase on top of it -- what's your reasoning for splitting this to exclude `sysfs_path` here, and similarly not adding `target_os = "android"` to the `DeviceInfo::sysfs_path` getter?

Un-rooted Android doesn't allow sysfs, but everything in `DeviceInfo` comes from sysfs, so `list_devices` isn't expected to work there at all; that's why you added `from_fd`. I was assuming you added `target_os = "android"` in this file for use on rooted devices, or just consistency with Linux, in which case it seems like `sysfs_path` should be included. Alternatively, should we instead  `cfg` out `list_devices` and `DeviceInfo`  entirely?

Reply:
Good catch. Yeah this is something I missed.

I feel like the correct thing to do would be to add `target_os = "android"` to the getter, `fmt::Debug` impl etc. for consistency between linux and rooted devices. Then add some docs in `list_devices` noting that this will fail on unrooted android, with a link to `wrap_device` like what libusb does: https://libusb.sourceforge.io/api-1.0/group__libusb__lib.html#gga07d4ec54cf575d672ba94c72b3c0de7ca5534030a8299c85555aebfae39161ef7